docs/developer/CI.md

```diff
@@ -0,0 +1,410 @@
+`ARCH`, `OPT` and `COMMS`. `comment` is used to distinguish otherwise similar
+variants and uniquely identify a particular configuration as well as apply
+modifications to the build (see [comment tags](#comment-tags)).
+
+It should be noted, however, that only certain combinations of variables are
+available on certain agents. For a full set of available combinations it is
+necessary to look into the `Jenkinsfile` itself as this list is frequently
+updated and changed.
+
+### Build Artefacts
+
+During a Jenkins run, various files will be preserved, making them available to
+download.
+
+Currently the artefacts that are preserved are:
+
+- Coverage reports
+- Linting reports
+- Failed test data
+- FORD Documentation
+
+### Reports
+
+During a Jenkins run, several "reports" may be created and made available in the
+side-bar of a pipeline that generates them.
+
+##### Linting and Compiler Warnings
+
+Following a build, Jenkins will collect the compiler warnings issued during a
+build along with, if a linting phase is run, the resulting linter issues into a
+report accessible through the side-bar.
+
+!!! warning
+
+    The warning collection will take the set of warnings from the first build using
+    any of the following compilers:
+
+    - gfortran
+    - ifort
+    - nag
+
+    This means that if the first item for each compiler in `builds_to_run` (see
+    [manual parametrised runs](#manual-parametrised-runs)) is not an `opt=debug`
+    build, you may miss most warnings.
+
+The reports allow the types of issues to be selected and filtered:
+
+- They can be grouped and sorted by selecting the appropriate buttons above the
+  main table and table headers.
+
+- The doughnut of warning severity is actually a clickable filter to filter to
+  warnings of that level!
+
+- The `+` icon to the left of the issue in details mode allows you to expand the
+  issue to get more detail.
+
+!!! failure "Known issue"
+
+    Due to the relatively large sizes of CASTEP source files, while selecting the
+    `file:line` link will take you to an annotated copy of the source at the point
+    of the issue in question, the rendering engine tends to bog down machines
+    causing browsers to struggle or fail to load the annotated file.
+
+##### FORD API Documentation
+
+As part of the full build of the `default` branch following a merge, Jenkins
+will trigger a [`FORD`](https://forddocs.readthedocs.io/en/latest/) API
+documentation build. The results of this build are accessible from the
+`docs-trigger` pipeline in the side-bar.
+
+##### Test Coverage Report
+
+Whenever a `coverage` build is run (i.e. with either the `ifort` or `gfortran`
+compiler and `opt=coverage`), Jenkins will accumulate the coverage reports and
+make them accessible in the side-bar.
+
+!!! failure "Known issue"
+
+    Due to current security settings on ANVIL the `ifort` reports are not
+    viewable through this view and must be downloaded to be viewed.
```

Comment:
Can we please explain how?  This is counterintuitive in the extreme.  I only just discovered how to do this myself a couple of days ago.